addons/l10n_in/models/account_invoice.py

```diff
@@ -66,6 +68,118 @@ def _compute_l10n_in_state_id(self):
+                )
+            else:
+                line.l10n_in_line_warning = True
+
+        def filter_warning_sections(move, warning_sections):
+            filtered_sections_tcs, filtered_sections_tds = [], []
+            for line in move.invoice_line_ids.filtered(
+                lambda l: (
+                    l.l10n_in_tds_tcs_section not in l.tax_ids.mapped('tax_group_id')
+                    and (l.credit or l.debit)
+                )
+            ):
+                if line.l10n_in_tds_tcs_section in warning_sections:
+                    section_name = line.l10n_in_tds_tcs_section.name[5:]
+                    if line.l10n_in_tds_tcs_section.l10n_in_is_tcs and section_name not in filtered_sections_tcs:
+                        filtered_sections_tcs.append(section_name)
+                    elif line.l10n_in_tds_tcs_section.l10n_in_is_tds and section_name not in filtered_sections_tds:
+                        filtered_sections_tds.append(section_name)
+            return filtered_sections_tcs, filtered_sections_tds
+
+        for move in self:
+            if move.country_code == 'IN' and move.state == 'posted':
+                warning_sections = []
+                per_transaction_total = move._l10n_in_calculate_per_transaction_total()
+                tax_fiscal_yearly_totals, tax_monthly_totals = move._l10n_in_calculate_aggregate_totals(
+                    move.is_entry()
+                    and move.invoice_line_ids.mapped('partner_id')
+                    or move.partner_id
+                )
+
+                # per transaction limit check
+                for tax_group_id, limits in per_transaction_total.items():
+                    for limit in limits:
+                        if (
+                            tax_group_id.l10n_in_is_per_transaction_limit
+                            and limit > tax_group_id.l10n_in_per_transaction_limit
+                            and is_warning_applicable(tax_group_id)
+                        ):
+                            warning_sections.append(tax_group_id)
+
+                # aggregate limit check
+                warning_sections.extend(check_aggregate_limits(tax_fiscal_yearly_totals))
+                warning_sections.extend(check_aggregate_limits(tax_monthly_totals))
+
+                for line in move.invoice_line_ids:
+                    tax_group = line.l10n_in_tds_tcs_section
+                    section = tax_group
+                    line_price = line._l10n_in_compute_line_per_transaction_amount()
+                    if (
+                        line.move_id.move_type in ['out_invoice', 'in_invoice', 'entry']
+                        and tax_group
+                        and tax_group not in line.tax_ids.mapped('tax_group_id')
+                        and (line.credit or line.debit)
+                    ):
+                        set_line_warning(line, tax_group, line_price, warning_sections, section)
```

Comment:
```suggestion
                        set_line_warning(line, tax_group, line_price, warning_sections)
```
I can't see point of sending `section = tax_group` any where in between